docs/book/src/cronjob-tutorial/testdata/project/internal/controller/cronjob_controller.go

```diff
@@ -500,7 +536,7 @@ func (r *CronJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	// +kubebuilder:docs-gen:collapse=constructJobForCronJob
 
 	// actually make the job...
-	job, err := constructJobForCronJob(&cronJob, missedRun)
+	job, err := constructJobForCronJob(cronJob, missedRun)
```

Comment:
Hi @lorenzofelletti

In a way that I would do that is: (see that this task is not so trivial, you need attention to details and use a lot the IDE to compare)

- Do the changes in the testdata samples directly as you are doing
- It is possible compare the code implementation of the book samples with the samples under [testdata/p](https://github.com/kubernetes-sigs/kubebuilder/tree/master/testdata/project-v4-with-plugins) see the code for Memcached and busybox which uses the deploy image plugin and is generated by: https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/testdata/generate.sh

Then, you can push a PR against your on fork.
See that the samples are tested in the CI here: https://github.com/kubernetes-sigs/kubebuilder/blob/master/.github/workflows/test-e2e-book.yml


If this workflow passes, then you can use the code changes.
Note that the test data check will fail because we generate the code of the book samples by running the commands and injecting the code on top. The whole code implementation for it is here: https://github.com/kubernetes-sigs/kubebuilder/tree/master/hack/docs/internal


When we call make generate-docs, the scripts are called, the projects generated, and the changes made afterwards; it ensures that the docs are always updated with the latest changes. 

But we can do it step by step.

Reply:
Hi @camilamacedo86!

Thanks for taking the time to write these instructions. I'll follow them and come back when I have something to share.